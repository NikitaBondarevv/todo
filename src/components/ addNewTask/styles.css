.addTask {
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  text-decoration: underline;
  box-shadow: 0px 1px 7px #000;
  margin: 0 0 14px 59px;
}

.editableText {
  height: 45px;
  width: 234px;
  font-size: 22px;
}
