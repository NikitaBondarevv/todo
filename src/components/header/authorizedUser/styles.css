.user {
  position: relative;
}

.userName {
  display: block;
  margin: 0 0 0 255px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
  background: url(./images/user.png) no-repeat 60px 0;
  min-width: 90px;
  text-decoration: none;
}

.userName:hover {
  color: #060e9a;
  background: url(./images/hoverUser.png) no-repeat 60px 0;
  text-decoration: underline;
}

.menu {
  background: rgba(255,255,255, 0.3);
  flex-direction: column;
  padding: 5px 17px 0;
  position: absolute;
  right: -5px;
  top: 26px;
  display: none;
}

.user:hover .menu {
  display: flex;
}

.menu a {
  text-decoration: none;
  color: #fff;
  font-size: 26px;
  line-height: 24px;
  white-space: nowrap;
  margin: 0 0 5px;
}

.menu a:hover {
  color: rgb(169, 169, 169);
}
