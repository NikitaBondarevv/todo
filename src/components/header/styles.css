header {
  font-size: 26px;
  line-height: 26px;
  display: grid;
  grid-template-columns: 0.1fr 1fr auto;
  align-items: center;
  padding: 23px 0 0;
  margin: 0 0 60px;
}

header .list {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  column-gap: 68px;
}

.list a {
  color: #fff;
  text-decoration: none;
  position: relative;
}

.list a:hover {
  color: #060e9a;
  text-decoration: underline;
}

.list a::before {
  background: url(./images/menu.png) no-repeat;
  content: "";
  display: block;
  width: 25px;
  height: 25px;
  right: -30px;
  bottom: 2px;
  position: absolute;
}

.list a:hover::before {
  background: url(./images/hoverMenu.png) no-repeat;
  content: "";
  display: block;
  width: 25px;
  height: 25px;
  right: -30px;
  bottom: 2px;
  position: absolute;
}

.list .home:hover::before {
  background-position: 2px 2px;
}

.list .taskList:hover::before {
  background-position: -23px 2px;
}

.list .contacts:hover::before {
  background-position: -46px 2px;
}

.list .taskList::before {
  background-position: -25px 0;
}

.list .contacts::before {
  background-position: -52px 0;
}

.createUser {
  margin: 0 0 0 220px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
  text-decoration: none;
  background: url(./images/user.png) no-repeat;
  background-position: 140px 0;
  width: 165px;
}

.createUser:hover {
  color: #060e9a;
  background: url(./images/hoverUser.png) no-repeat;
  background-position: 140px 0;
  text-decoration: underline;
}
