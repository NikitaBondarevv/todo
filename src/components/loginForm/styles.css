.loginForm {
  display: flex;
  flex-direction: column;
  width: 675px;
  height: 406px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 80px;
}

.loginForm input,
.loginForm button[type=submit] {
  font: 24px/24px "MyriadProLight";
  padding: 5px 5px 5px 30px;
  width: 247px;
  height: 55px;
  border-radius: 7px;
  margin: 0 0 15px 0;
  border: 1px solid #fff;
  font-style: italic;
  color: #000;
}

.loginForm input::placeholder {
  color: #000;
}

.loginForm button[type=submit] {
  padding: 0;
  margin: 0;
  background: #0ff;
  color: #fff;
  height: 53px;
  width: 143px;
  border: 1px solid #b1bad8;
  box-shadow: 0 0 5px #b1bad8;
}

.loginForm button[type=submit]:disabled {
  background: #316262;
}

.password {
  position: relative;
}

.loginForm .password::before {
  background: url(./images/form-elements.png) no-repeat;
  content: "";
  width: 17px;
  position: absolute;
  z-index: 2;
  left: 250px;
  top: 23px;
}

.loginForm .password::before {
  background-position: 0 -33px;
  height: 18px;
}

.error {
  color: red;
  font-size: 25px;
}
