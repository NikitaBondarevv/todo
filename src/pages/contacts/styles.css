.contacts {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 0 0;
  text-align: center;
  background: rgb(7 21 147 / 50%);
  width: 835px;
  color: #fff;
  font: 35px/40px "MyriadProRegular";
  margin: 0 auto 100px;
  letter-spacing: 1px;
  height: 487px;
}

.email {
  background: none;
  color: #fff;
  font-weight: bold;
  text-decoration: none;
  padding: 0 0 0 5px;
}

.email:hover {
  text-decoration: underline;
}
