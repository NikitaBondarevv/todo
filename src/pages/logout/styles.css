.logout {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 29px 104px;
  background: rgb(7 21 147 / 50%);
  width: 777px;
  height: 383px;
  color: #fff;
  font: 24px/40px "MyriadProRegular";
  margin: 0 auto 100px;
  letter-spacing: 1px;
}

.error {
  color: red;
  font-size: 25px;
  margin: 0 0 0 5px;
}
