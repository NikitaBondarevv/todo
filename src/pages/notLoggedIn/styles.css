.content {
  text-align: center;
  width: 775px;
  height: 358px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 30px 108px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
}

.content p {
  margin: 0;
  letter-spacing: 1px;
  margin: 0 0 160px;
}

.links {
  display: flex;
  justify-content: center;
  column-gap: 30px;
}

.link {
  display: block;
  color: #000;
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  box-shadow: 0px 1px 7px #000;
  text-decoration: none;
}

.link:hover {
  text-decoration: underline;
}
