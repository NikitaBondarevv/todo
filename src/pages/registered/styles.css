.content {
  display: flex;
  flex-direction: column;
  width: 775px;
  height: 358px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 30px 108px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
}

.content p {
  margin: 0 0 40px;
  letter-spacing: 1px;
}

.link {
  display: block;
  color: #000;
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  box-shadow: 0px 1px 7px #000;
  margin: 0 0 14px 59px;
  text-decoration: none;
  margin: 156px 0 0 50px;
}

.link:hover {
  text-decoration: underline;
}
