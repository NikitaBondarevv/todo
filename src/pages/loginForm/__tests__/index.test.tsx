import { act, fireEvent, render } from '@testing-library/react'

import { login } from 'contracts/login'
import { LoginForm } from '..'
import styles from './styles.css'

jest.mock('contracts/login', () => ({
  login: jest.fn().mockReturnValue({
    email: 'test@test.com',
    password: 'test-password'
  })
}))

describe('<LoginForm />', () => {
  const loginMock = login as jest.Mock

  beforeEach(() => {
    jest.resetAllMocks()
    loginMock.mockReturnValue({
      email: 'test@test.com',
      password: 'test-password'
    })
  })

  const setUser = jest.fn()

  test('should match snapshot', () => {
    const { asFragment } = render(<LoginForm setUser={setUser} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should called setUser if not error', async () => {
    const { container } = render(<LoginForm setUser={setUser} />)
    const form = container.querySelector(`.${styles.loginForm}`)

    await act(() => fireEvent.submit(form!))

    expect(setUser).toHaveBeenCalled()
  })

  test('should show error window if there is error', async () => {
    const error = { error: 'test-error' }

    loginMock.mockReturnValueOnce(error)

    const { container } = render(<LoginForm setUser={setUser} />)
    const form = container.querySelector(`.${styles.loginForm}`)

    await act(() => fireEvent.submit(form!))

    expect(setUser).not.toHaveBeenCalled()
  })
})
