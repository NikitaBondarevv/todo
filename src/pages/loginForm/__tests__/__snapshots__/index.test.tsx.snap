// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="loginForm"
  >
    <input
      name="email"
      placeholder="Email"
      type="text"
    />
    <div
      class="password"
    >
      <input
        name="password"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
